client/src/app/components/forms/rsvp/rsvp.component.ts

```diff
@@ -21,7 +26,18 @@ export class RsvpFormComponent {
 		guests: this.fb.array([])
 	});
 
-	constructor(private fb: NonNullableFormBuilder){}
+	constructor(private fb: NonNullableFormBuilder, private route: ActivatedRoute, private api: RsvpService){}
+
+	ngOnInit() {
+		const codeInput = this.route.snapshot.queryParamMap.get('code');
```

Comment:
If you're up for a refactor (if not I understand lol) - I don't like the idea of calling the api from within here. I'd rather you pass it in as an Input().

Here's a [decent-ish article on passing info from a parent component to a child component](https://zeroesandones.medium.com/how-to-pass-data-from-parent-to-child-component-in-angular-9-dc7a0a64256c)

This would be the child component, and the RsvpComponent proper would be the parent component. They're a bit poorly named, but the idea is that the RsvpComponent is the whole page, and handles the routes and stuff.  The RsvpFormComponent only handles things relevant to the form - it shouldn't have access to the api.